src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/jpa/ActivityMigrationMapping.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.annotation.CreatedDate
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.Transient
+import org.springframework.data.domain.Persistable
+import java.time.LocalDateTime
+
+data class ActivityMigrationMapping(
+
+  @Id
+  val nomisCourseActivityId: Long,
+
+  val activityScheduleId: Long,
+
+  val activityScheduleId2: Long?,
+
+  @CreatedDate
+  val whenCreated: LocalDateTime? = null,
+
+  val label: String,
+
+  @Transient
+  @Value("false")
+  val new: Boolean = true,
+
+) : Persistable<Long> {
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other !is ActivityMapping) return false
+
+    if (nomisCourseActivityId != other.nomisCourseActivityId) return false
+
+    return true
```

Comment:
Is this following the pattern of other Entities? the equals fun is different  elsewhere, i.e. includes the 
 _if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false_
  line

Reply:
I copied from the original activities mapping entity which you did 😄. It's only an issue when we have updates - this service shouldn't have any updates - but I'll fix them both for consistency.

Reply:
Lol whoops, ta!

Reply:
Ah I know why now - we're not using Hibernate! It's all that reactive stuff.